sycl/test/fpga_tests/fpga_io_pipes.cpp

```diff
@@ -0,0 +1,132 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUNx: %ACC_RUN_PLACEHOLDER %t.out
+//==------------ fpga_io_pipes.cpp - SYCL FPGA pipes test ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+#include <CL/sycl/intel/fpga_extensions.hpp>
+#include <fstream>
+#include <iostream>
+
+#include "io_pipe_def.h"
+
+// TODO: run is disabled, since no support added in FPGA backend yet. Check
+// implementation correctness from CXX and SYCL languages perspective.
+
+// This test is supposed to be run only on Intel FPGA emulator. Change it when
+// we have more experience with IO pipe feature in SYCL.
+// The emulator creates files (one for I pipe, another for O pipe) with the
+// appropriate naming, where a data flowing through a pipe can be stored.
+// So in the test we need to create these files and use them appropriately.
+// The name is taken as IO pipe ID.
+const size_t InputData = 42;
+const std::string InputFileName = "0.txt";
+const std::string OutputFileName = "1.txt";
+
+void createInputFile(const std::string &filename) {
+  std::ofstream Input(filename);
+  if (Input.is_open()) {
+    Input << InputData;
+    Input.close();
+  }
+}
+
+int validateOutputFile(const std::string &filename) {
+  std::ifstream Output(filename);
+  std::string Line;
+  std::vector<size_t> Result;
+  if (Output.is_open()) {
+    // In the test we write the only number into the pipe, but a backend might
+    // have a bug of incorrect interpretetion of capacity of the pipe. In this
+    // case lets read all the lines of the output file to catch this it.
```

Comment:
```suggestion
    // case let's read all the lines of the output file to catch this.
```
Note I might be wrong since I am not a native English speaker...

Reply:
This one is totally legit comment =)